### PR TITLE
[TAR-1335][19.03] Add optional dependency on nftables for RHEL8+ derivatives

### PR DESCRIPTION
**Pull request by droopy4096**

### CODE REVIEW

**Comment by zelahi** [comment]
@droopy4096 should this be opened against master: https://github.com/docker/docker-ce-packaging/blob/master/rpm/SPECS/docker-ce.spec and then we backport it to this branch.  It would help when we are dealing with merge conflicts

**Comment by zelahi** [comment]
Another small knitpit.  Sorry I modified the title a bit.  I added the 19.03 as a mini heading.  It's so we an tell what branch this is being opened against =)

**Comment by droopy4096** [comment]
@zelahi now we have branch name in two places ;) I'll remove the suffix one...

File: rpm/SPECS/docker-ce.spec [hunk 1]
```diff
@@ -19,7 +19,11 @@ Requires: docker-ce-cli
 Requires: container-selinux >= 2:2.74
 Requires: libseccomp >= 2.3
 Requires: systemd
+%if 0%{?rhel} >= 8
+Requires: iptables or nftables
```
Comment (thaJeztah) [inline rpm/SPECS/docker-ce.spec]: I'm curious; I would've expected the `nftables` package to "provide" `iptables`, is that not the case?

Comment (droopy4096) [reply rpm/SPECS/docker-ce.spec]: I'll file the second PR for the master.

nope - nftables does *not* provide `iptables` 
```shell
$ rpm --provides -q nftables
config(nftables) = 1:0.9.0-8.el8
libnftables.so.0()(64bit)
nftables = 1:0.9.0-8.el8
nftables(x86-64) = 1:0.9.0-8.el8
```

**Comment by thaJeztah** [comment]
@droopy4096 I see you closed this one; are you working on a PR for master?

**Comment by droopy4096** [comment]
@thaJeztah  PR-414

**Pull request closed without merging**